File: src/main/java/com/gitlab/daring/image/command/transform/CropRectVarCommand.java
package com.gitlab.daring.image.command.transform;

import com.gitlab.daring.image.command.BaseCommand;
import com.gitlab.daring.image.command.CommandEnv;
import com.gitlab.daring.image.command.parameter.StringParam;
import org.bytedeco.javacpp.opencv_core.Rect;

import java.awt.*;

import static com.gitlab.daring.image.util.OpencvConverters.toOpencv;

public class CropRectVarCommand extends BaseCommand {

    final StringParam rectKey = stringParam("rect");
    final StringParam matKey = stringParam("");

    public CropRectVarCommand(String... args) {
        super(args);
    }

    @Override
    public void execute(CommandEnv env) {
        String mk = matKey.v;
        Rect rect = toOpencv((Rectangle) env.vars.get(rectKey.v));
        env.putMat(mk, env.getMat(mk).apply(rect));
    }

}

File: src/main/kotlin/com/gitlab/daring/image/command/transform/CropRectVarCommand.kt
package com.gitlab.daring.image.command.transform

import com.gitlab.daring.image.command.CommandEnv
import com.gitlab.daring.image.command.KBaseCommand
import com.gitlab.daring.image.util.ImageUtils.cropMat
import java.awt.Rectangle

internal class CropRectVarCommand(vararg params: String) : KBaseCommand(*params) {

    val rectKey = stringParam("rect");
    val matKey = stringParam("");

    override fun execute(env: CommandEnv) {
        val mk = matKey.v;
        val rect = env.getVar<Rectangle>(rectKey.v);
        env.putMat(mk, cropMat(env.getMat(mk), rect));
    }

}